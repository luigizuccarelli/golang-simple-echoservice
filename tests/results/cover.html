
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go (79.1%)</option>
				
				<option value="file1">/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package main

import (
        "encoding/json"
        "fmt"
        "github.com/dgrijalva/jwt-go"
        "net/http"
        "os"
        "strings"
)

const (
        CONTENTTYPE     string = "Content-Type"
        APPLICATIONJSON string = "application/json"
)

type Claims struct {
        jwt.StandardClaims
}

func MiddlewareAuth(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        var response Response

        token := r.Header.Get(strings.ToLower("Authorization"))

        if token == "" </span><span class="cov8" title="1">{
                w.WriteHeader(http.StatusUnauthorized)
                response = Response{Code: 403, StatusCode: "403", Status: "ERROR", Message: "Forbidden", Payload: SchemaInterface{}}
        }</span> else<span class="cov8" title="1"> {

                // Remove Bearer
                tknStr := strings.Trim(token[7:], " ")
                logger.Debug(fmt.Sprintf("Token : %s", tknStr))
                addHeaders(w, r)
                handleOptions(w, r)

                // Initialize a new instance of `Claims`
                claims := &amp;Claims{}

                secret := os.Getenv("JWT_SECRETKEY")
                logger.Trace(fmt.Sprintf("JWT SECRET : %s", secret))

                var jwtKey = []byte(secret)

                // Parse the JWT string and store the result in `claims`.
                // Note that we are passing the key in this method as well. This method will return an error
                // if the token is invalid (if it has expired according to the expiry time we set on sign in),
                // or if the signature does not match
                tkn, err := jwt.ParseWithClaims(tknStr, claims, func(token *jwt.Token) (interface{}, error) </span><span class="cov0" title="0">{
                        return jwtKey, nil
                }</span>)

                <span class="cov8" title="1">if err != nil </span><span class="cov8" title="1">{
                        if err.Error() == jwt.ErrSignatureInvalid.Error() </span><span class="cov0" title="0">{
                                w.WriteHeader(http.StatusUnauthorized)
                                response = Response{Code: 403, StatusCode: "403", Status: "ERROR", Message: "Forbidden", Payload: SchemaInterface{}}
                        }</span> else<span class="cov8" title="1"> {
                                w.WriteHeader(http.StatusBadRequest)
                                response = Response{Code: 400, StatusCode: "400", Status: "ERROR", Message: "Bad Request", Payload: SchemaInterface{}}
                        }</span>
                } else<span class="cov0" title="0"> {
                        if !tkn.Valid </span><span class="cov0" title="0">{
                                w.WriteHeader(http.StatusUnauthorized)
                                response = Response{Code: 403, StatusCode: "403", Status: "ERROR", Message: "Forbidden", Payload: SchemaInterface{}}
                        }</span> else<span class="cov0" title="0"> {
                                response = Response{Code: 200, StatusCode: "200", Status: "OK", Message: "Data uploaded succesfully", Payload: SchemaInterface{}}
                                w.WriteHeader(http.StatusOK)
                        }</span>
                }
        }
        <span class="cov8" title="1">b, _ := json.MarshalIndent(response, "", "        ")
        logger.Debug(fmt.Sprintf("AuthHandler response : %s", string(b)))
        fmt.Fprintf(w, string(b))</span>
}

func IsAlive(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        fmt.Fprintf(w, "{ \"version\" : \""+os.Getenv("VERSION")+"\" , \"name\": \"Auth\" }")
}</span>

// headers (with cors) utility
func addHeaders(w http.ResponseWriter, r *http.Request) <span class="cov8" title="1">{
        var request []string
        for name, headers := range r.Header </span><span class="cov8" title="1">{
                name = strings.ToLower(name)
                for _, h := range headers </span><span class="cov8" title="1">{
                        request = append(request, fmt.Sprintf("%v: %v", name, h))
                }</span>
        }

        <span class="cov8" title="1">logger.Trace(fmt.Sprintf("Headers : %s", request))

        w.Header().Set(CONTENTTYPE, APPLICATIONJSON)
        // use this for cors
        w.Header().Set("Access-Control-Allow-Origin", "*")
        w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
        w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")</span>

}

// simple options handler
func handleOptions(w http.ResponseWriter, r *http.Request) bool <span class="cov8" title="1">{
        if r.Method == "OPTIONS" </span><span class="cov0" title="0">{
                return true
        }</span>
        <span class="cov8" title="1">return false</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package main

import (
        "errors"
        "fmt"
        "os"
        "strconv"
        "strings"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        logger.Trace(fmt.Sprintf("name %s : required %t", name, required))
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return errors.New(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span> else<span class="cov8" title="1"> {
                        logger.Error(fmt.Sprintf("%s envar is empty please set it", name))
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
// Each microservice will obviously have a diffefrent envars so change where needed
func ValidateEnvars() error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "NAME,false",
                "SERVER_PORT,true",
                "JWT_SECRETKEY,true",
                "VERSION,true",
        }
        for x, _ := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x]); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
