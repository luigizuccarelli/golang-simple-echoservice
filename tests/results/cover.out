mode: set
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:21.61,26.17 3 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:71.2,73.27 3 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:26.17,29.3 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:29.8,49.95 9 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:53.3,53.17 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:49.95,51.4 1 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:53.17,54.54 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:54.54,57.5 2 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:57.10,60.5 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:61.9,62.18 1 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:62.18,65.5 2 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:65.10,68.5 2 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:76.54,78.2 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:81.57,83.38 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:90.2,96.134 5 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:83.38,85.29 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:85.29,87.4 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:101.65,102.27 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:105.2,105.14 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/middleware.go:102.27,104.3 1 0
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:12.36,16.27 4 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:24.2,24.12 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:16.27,17.15 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:17.15,20.4 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:20.9,22.4 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:30.29,38.26 2 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:43.2,43.12 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:38.26,39.46 1 1
/home/lzuccarelli/Projects/myportfolio/myportfolio-authinterface/validate.go:39.46,41.4 1 1
